vip-helpers/vip-non-production.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+if ( ! defined( 'VIP_GO_APP_ENVIRONMENT' ) || 'production' === constant( 'VIP_GO_APP_ENVIRONMENT' ) ) {
+	return;
+}
+
+add_action( 'wp_footer', 'vip_do_non_prod_bar', 10000 );
+add_action( 'admin_footer', 'vip_do_non_prod_bar', 10000 );
+
+function vip_do_non_prod_bar() {
+	if ( ! current_user_can( 'edit_posts' ) ) {
+		return;
+	}
+
+	if ( apply_filters( 'vip_show_non_prod_bar', true ) ) :
+		?>
+		<div id="vip-non-prod-bar">
+			<span>
+				<strong><?php echo esc_html( constant( 'VIP_GO_APP_ENVIRONMENT' ) ); ?></strong>
+			</span>
+		</div>
+		<script>
+			const nonProdBar = document.querySelector('#vip-non-prod-bar');
+			nonProdBar.addEventListener( 'click', function() {
+				this.classList.toggle('which-env');
+			} );
+
+			const sandboxedBar = document.querySelector('#wpcom-sandboxed-bar');
+			const debugBar = document.querySelector('#a8c-debug-flag');
+			if ( ( sandboxedBar && debugBar ) || ( ! sandboxedBar && debugBar ) ) {
```

Comment:
If your above condition is correct, then it is equivalent to:

```suggestion
			if ( debugBar ) {
```

```
(A && B) || (!A && B) = B

!!((A && B) || (!A && B)) =
= !(!(A && B) && !(!A && B)) =
= !((!A || !B) && (A || !B)) =
= !(0 || A && !B || !A && !B || !B) =
= !((A && !B) || (!A && !B) || !B) =
= (!A || B) && (A || B) && B =
= (0 || AB || B) && B =
= A && B || B =
= B
```

Wow, I still remember discrete math 😀

Reply:
Oops, definitely can reduce it 🙂 thanks!